nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -333,7 +335,8 @@ private void setProxy(OkHttpClient.Builder builder) {
             final Record record;
             try (final InputStream is = responseBody.byteStream();
                 final InputStream bufferedIn = new BufferedInputStream(is)) {
-                record = handleResponse(bufferedIn, responseBody.contentLength(), context);
+                String toString = IOUtils.toString(bufferedIn, StandardCharsets.UTF_8);
+                record = handleResponse(new ByteArrayInputStream(toString.getBytes(StandardCharsets.UTF_8)), responseBody.contentLength(), context);
```

Comment:
@MikeThomsen, I didn't notice this change before, but it seems problematic. Reading in the entire response body to a string, and then converting to a byte array before passing to `handleResponse()` could introduce memory issues. Was there anything in particular that prompted this change, or can it be reverted?

This appears to be the only non-test file changed, so everything else looks good.

Reply:
Although this may only be reading one Record, the `UTF-8` encoding is also a concern, versus just passing the InputStream directly to the Record Reader.